voltage/messageable.py

```diff
@@ -280,15 +280,23 @@ async def purge(self, amount: int):
             The amount of messages to purge.
         """
         channel_id = await self.get_id()
-        for i in await self.cache.http.fetch_messages(channel_id, "Latest", limit=amount):
-            try:
-                await self.cache.http.delete_message(channel_id, i["_id"])
-            except HTTPError as e:
-                status = e.response.status
-                if status == 404:
-                    pass
-                else:
-                    raise
+        message_list = []
+        [
+            message_list.append(message["_id"])
+            for message in await self.cache.http.fetch_messages(
+                channel_id, "Latest", limit=amount
+            )
+        ]
+        try:
+            await self.cache.http.bulk_delete_messages(
```

Comment:
you should probably chunk this as to not try to clear more than 100 messages at once

Reply:
Changed in most recent commit